subworkflows/nf-core/gatk4/joint_germline_variant_calling/main.nf

```diff
@@ -1,92 +1,172 @@
+                                            new_meta = [id: "recalibrated_joint_variant_calling",
+                                                        patient: "all_samples",
+                                                        variantcaller: "haplotypecaller",
+                                                        num_intervals: meta.num_intervals]
+
+                                            [new_meta, vcf, tbi, recal, index, tranche]
+                                        }
+
+    GATK4_APPLYVQSR_SNP(vqsr_input_snp,
+                        fasta,
+                        fai,
+                        dict )
+
+    GATK4_APPLYVQSR_INDEL(vqsr_input_indel,
+                        fasta,
+                        fai,
+                        dict )
+
+    vqsr_snp_vcf = GATK4_APPLYVQSR_SNP.out.vcf
+    vqsr_indel_vcf = GATK4_APPLYVQSR_INDEL.out.vcf
 
-    //     ch_versions   = ch_versions.mix(APPLYVQSR.out.versions)
+    //
+    //Merge VQSR outputs into final VCF
+    //
+    MERGE_VQSR(
+        vqsr_snp_vcf.mix(vqsr_indel_vcf).groupTuple(),
+        dict
+    )
+    ch_versions = ch_versions.mix(GATK4_GENOMICSDBIMPORT.out.versions,
+                              GATK4_GENOTYPEGVCFS.out.versions,
+                              VARIANTRECALIBRATOR_SNP.out.versions,
+                              GATK4_APPLYVQSR_SNP.out.versions
+                             )
 
-    // }
 
     emit:
-    // haplotc_vcf    = run_haplotc ? HAPLOTYPECALLER.out.vcf.collect()          : [] // channel: [ val(meta), [ vcf ] ]
-    // haplotc_index  = run_haplotc ? HAPLOTYPECALLER.out.tbi.collect()          : [] // channel: [ val(meta), [ tbi ] ]
-
-    // genomicsdb     =               GENOMICSDBIMPORT.out.genomicsdb.collect()       // channel: [ val(meta), [ genomicsdb ] ]
-
-    // genotype_vcf   =               GENOTYPEGVCFS.out.vcf.collect()                 // channel: [ val(meta), [ vcf ] ]
-    // genotype_index =               GENOTYPEGVCFS.out.vcf.collect()                 // channel: [ val(meta), [ tbi ] ]
-
-    // recal_file     = run_vqsr    ? VARIANTRECALIBRATOR.out.recal.collect()    : [] // channel: [ val(meta), [ recal ] ]
-    // recal_index    = run_vqsr    ? VARIANTRECALIBRATOR.out.idx.collect()      : [] // channel: [ val(meta), [ idx ] ]
-    // recal_tranches = run_vqsr    ? VARIANTRECALIBRATOR.out.tranches.collect() : [] // channel: [ val(meta), [ tranches ] ]
-
-    // vqsr_vcf       = run_vqsr    ? APPLYVQSR.out.vcf.collect()                : [] // channel: [ val(meta), [ vcf ] ]
-    // vqsr_index     = run_vqsr    ? APPLYVQSR.out.tbi.collect()                : [] // channel: [ val(meta), [ tbi ] ]
-
-    versions       =               ch_versions                                     // channel: [ versions.yml ]
+    versions       = ch_versions                               // channel: [ versions.yml ]
+    genotype_vcf   = Channel.empty().mix(vcfs_sorted_input.no_intervals,
```

Comment:
the vcfs_sorted_input.no_intervals also needs the variantcaller: "haplotypecaller" in it smeta map here to make sure annotation is placing it in the proper folder